diff -uprN ./mruby-1.2.0/build_config.rb mruby-1.2.0-patched/build_config.rb
--- ./mruby-1.2.0/build_config.rb	2016-06-06 21:08:24 +0900
+++ mruby-1.2.0-patched/build_config.rb	2016-06-06 21:08:07 +0900
@@ -19,6 +19,20 @@
   # conf.gem :github => 'masuidrive/mrbgems-example', :checksum_hash => '76518e8aecd131d047378448ac8055fa29d974a9'
   # conf.gem :git => 'git@github.com:masuidrive/mrbgems-example.git', :branch => 'master', :options => '-v'
 
+  # Use mruby-require and mruby-tempfile
+  conf.gem :git => 'https://github.com/iij/mruby-tempfile' do |g|
+    if ENV['VisualStudioVersion'] || ENV['VSINSTALLDIR']
+      p Dir.pwd
+      sh "patch -p1 < ../../../patches/mruby-tempfile-vc14.patch"
+    end
+  end
+  conf.gem :git => 'https://github.com/iij/mruby-require' do |g|
+    if ENV['VisualStudioVersion'] || ENV['VSINSTALLDIR']
+      p Dir.pwd
+      sh "patch -p1 < ../../../patches/mruby-require-vc14.patch"
+    end
+  end
+
   # include the default GEMs
   conf.gembox 'default'
   # C compiler settings
