diff -uprN ./mruby-1.2.0/build/mrbgems/mruby-tempfile/src/tempfile.c mruby-1.2.0.patched/build/mrbgems/mruby-tempfile/src/tempfile.c
--- ./mruby-1.2.0/build/mrbgems/mruby-tempfile/src/tempfile.c	2016-06-06 21:16:58 +0900
+++ mruby-1.2.0.patched/build/mrbgems/mruby-tempfile/src/tempfile.c	2016-06-06 21:32:04 +0900
@@ -12,7 +12,18 @@
 
 //#include <stdlib.h>
 #include <sys/types.h>
+#if defined(_MSC_VER)
+#include <io.h>
+#define open(filename, oflag) _open(filename, oflag)
+#define close(fd) _close(fd)
+#define access(path, mode) _access(path, mode)
+#define unlink(filename) _unlink(filename)
+#define F_OK 00
+#include <process.h>
+#define getpid() _getpid()
+#else
 #include <unistd.h>
+#endif
 
 static void mrb_tempfile_path_free(mrb_state *, void *);
 
