diff -uprN ./mruby-1.2.0/build/mrbgems/mruby-require/src/require.c mruby-1.2.0.patched/build/mrbgems/mruby-require/src/require.c
--- ./mruby-1.2.0/build/mrbgems/mruby-require/src/require.c	2016-06-06 19:42:40 +0900
+++ mruby-1.2.0.patched/build/mrbgems/mruby-require/src/require.c	2016-06-06 19:41:00 +0900
@@ -3,7 +3,13 @@
 #endif
 #include <fcntl.h>
 #include <setjmp.h>
+#if defined(_MSC_VER)
+#include <io.h>
+#define open(filename,oflag) _open(filename,oflag)
+#define close(fd) _close(fd)
+#else
 #include <unistd.h>
+#endif
 
 #include "mruby.h"
 #include "mruby/compile.h"
@@ -109,7 +115,9 @@
 #else
   char tmpname[] = "tmp.XXXXXXXX";
 #endif
+#if !defined(_MSC_VER)
   mode_t mask;
+#endif
   FILE *tmpfp = NULL;
   int fd = -1, ret;
   mrb_irep *irep;
@@ -121,12 +129,16 @@
   }
   path_ptr = mrb_str_to_cstr(mrb, path);
 
+#if !defined(_MSC_VER)
   mask = umask(077);
+#endif
   fd = mkstemp(tmpname);
   if (fd == -1) {
     mrb_sys_fail(mrb, "can't create mkstemp() at mrb_require_load_rb_str");
   }
+#if !defined(_MSC_VER)
   umask(mask);
+#endif
 
   tmpfp = fdopen(fd, "r+");
   if (tmpfp == NULL) {
